tests/js/timeout-4.js

```diff
@@ -0,0 +1,11 @@
+// The task stops after `main` exits and then we flush the micro-task queue. Any
+// promise that hasn't settled is abandoned. Therefore, the error after the
+// timeout will not trigger our unhandled-rejected-promises task killing because
+// it will never execut and throw.
+
+async function main() {
+  // No await.
```

Comment:
Can you add the full comment on why here, too? If we look at this file in isolation later, this'll be confusing.